packages/eslint-plugin/test/no-deprecated-components.test.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// tslint:disable object-literal-sort-keys
+/* eslint-disable no-template-curly-in-string */
+
+import { TSESLint } from "@typescript-eslint/utils";
+import dedent from "dedent";
+
+import { noDeprecatedComponentsRule } from "../src/rules/no-deprecated-components";
+
+const ruleTester = new TSESLint.RuleTester({
+    parser: require.resolve("@typescript-eslint/parser"),
+    parserOptions: {
+        ecmaFeatures: {
+            jsx: true,
+        },
+        sourceType: "module",
+    },
+});
+
+console.info("Testing no-deprecated-components rule...");
+ruleTester.run("no-deprecated-components", noDeprecatedComponentsRule, {
+    invalid: [
+        {
+            code: dedent`
+                import { CollapsibleList } from "@blueprintjs/core";
+
+                return <CollapsibleList />
+            `,
+            errors: [
+                { messageId: "nonDeprecated", data: { deprecated: "CollapsibleList", nonDeprecated: "OverflowList" } },
+            ],
+        },
+        {
+            code: dedent`
+                import * as BP from "@blueprintjs/core";
+
+                return <BP.CollapsibleList />
+            `,
+            errors: [
+                { messageId: "nonDeprecated", data: { deprecated: "CollapsibleList", nonDeprecated: "OverflowList" } },
+            ],
+        },
+        {
+            code: dedent`
+                import { AbstractComponent } from "@blueprintjs/core";
+
+                export class MyClass extends AbstractComponent {
+                }
+            `,
+            errors: [
+                {
+                    messageId: "nonDeprecated",
+                    data: { deprecated: "AbstractComponent", nonDeprecated: "AbstractComponent2" },
+                },
+            ],
+        },
+        {
+            code: dedent`
+                import * as BP from "@blueprintjs/core";
+
+                export class MyClass extends BP.AbstractComponent {
+                }
+                class MyClass2 extends BP.AbstractComponent {}
```

Comment:
looks like two errors should be expected here instead of just one

Reply:
hm.. it passes both if i pass 1 error or 2 errors below 🤔